src/Constraint/TypeConstraint.php

```diff
@@ -80,21 +84,15 @@ public function normalize(stdClass $schema, Context $context, Walker $walker)
      */
     public function apply($instance, stdClass $schema, Context $context, Walker $walker)
     {
+        $actualType = gettype($instance);
+
         if (is_string($schema->type)) {
-            if (!Types::isA($instance, $schema->type)) {
+            if (!isset(Types::$phpJsonCompatibility[$actualType.$schema->type])) {
                 $context->addViolation('instance must be of type %s', [$schema->type]);
             }
-        } else {
-            $hasMatch = false;
-
-            foreach ($schema->type as $type) {
-                if (Types::isA($instance, $type)) {
-                    $hasMatch = true;
-                    break;
-                }
-            }
 
-            if (!$hasMatch) {
+        } else {
+            if (!isset(Types::$phpToJson[$actualType]) || !in_array(Types::$phpToJson[$actualType], $schema->type)) {
```

Comment:
This is currently handled in normalize phase, see https://github.com/stefk/JVal/pull/10/commits/b2f7fd7dcc4903769429fa70d88ea1512ecfaa12#diff-96dad29cd96ce01c1dbbd358fd2594e5R65.